src/microbe_stage/MicrobeStage.cs

```diff
@@ -235,10 +235,10 @@ public void SetupStage()
         if (Settings.Instance == null)
             GD.PrintErr("Settings load problem");
 
+        spawner.Init(Clouds);
```

Comment:
It looks like you added extra stuff for the spawn system to keep track of. However, you didn't remove the PatchManager also needing those resources. I think you should re-design this part so that the Spawn system does *not* know about these kinds of details, so this and the SetCurrentGame would be removed in favour of the PatchManager sending only the needed info to the SpawnSystem.